src/scripts/quests/questTypes/GainMoneyQuest.ts

```diff
@@ -8,13 +8,17 @@ class GainMoneyQuest extends Quest implements QuestInterface {
     }
 
     public static generateData(): any[] {
-        const amount = SeededRand.intBetween(20000, 60000);
-        const reward = this.calcReward(amount);
-        return [amount, reward];
+        const availableGyms = GameConstants.RegionGyms[player.highestRegion()].filter(x => App.game.badgeCase.hasBadge(GymList[x].badgeReward));
+        const maxGym = GymList[availableGyms[availableGyms.length - 1]];
```

Comment:
It's possible the player doesn't have any badges from their highest region yet, which makes this crash